### PR TITLE
Add tests processing headers json data converter

### PR DESCRIPTION
**Pull request by AdrianMontaguSmartDCSIT**
This PR updates json_data_converter:
- Updated type hints
- Split navigate_segment function into navigate_list and navigate_dict.
- Add build_nested_dict function to extract_nested_value function.

Tests have been written for the following functions:

- navigate_dict
- navigate_list
- extract_simple_value
- extract_value_from_path
- update_header_dict
- build_nested_dict

### CODE REVIEW

File: tel2puml/find_unique_graphs/otel_ingestion/json_data_converter.py [hunk 1]
```diff
@@ -629,104 +631,148 @@ def process_header(
...
+) -> dict[str, Any]:
+    """Navigate through a list of the JSON data.
+
+    :param data: The current data object
+    :type data: `list`[`dict`[`str`, `Any`]]
+    :return: The value at index 0 of the list
+    :rtype: `dict`[`str`, `Any`]
+    """
+    try:
+        return data[0]
```
Comment (FreddieMatherSmartDCSIT) [inline tel2puml/find_unique_graphs/otel_ingestion/json_data_converter.py:754]: why are we only taking the first entry from the list -will we not be skipping data?

File: tests/tel2puml/find_unique_graphs/test_json_data_converter.py [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+            _navigate_dict(data4, segment4)
+
+        # Test invalid type
+        data5 = {"key1": 123}  # Not a dict, list, or str
+        segment5 = "key1"
+        with pytest.raises(TypeError):
+            _navigate_dict(data5, segment5)
+
+    @staticmethod
+    def test_navigate_list() -> None:
```
Comment (FreddieMatherSmartDCSIT) [inline tests/tel2puml/find_unique_graphs/test_json_data_converter.py:57]: what about a list with more than one entry?

**Review by FreddieMatherSmartDCSIT — CHANGES_REQUESTED** [review]
Unsure why when looking at a list we are only looking at the first entry. Couldn't it not be in the first entry?

File: tests/tel2puml/find_unique_graphs/test_json_data_converter.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+        assert _build_nested_dict(path_segments3, parsed_data3) == expected3
+
+        # Test error handling with one path segment. In reality there will
+        # always be more than one path segment since the path requires '::'
+        # for the function to be reached
+        path_segments4 = ["key1"]
+        parsed_data4 = 123
+        with pytest.raises(TypeError):
+            _build_nested_dict(
+                path_segments4, parsed_data4  # type: ignore[arg-type]
```
Comment (FreddieMatherSmartDCSIT) [inline tests/tel2puml/find_unique_graphs/test_json_data_converter.py:225]: Can just make the above astring so we don't have to ignore the mypy error

**Review by FreddieMatherSmartDCSIT — CHANGES_REQUESTED** [review]
Just change the type of the input on the test so we don't get a mypy error

File: tests/tel2puml/find_unique_graphs/test_json_data_converter.py (hunk 3 above)
Comment (AdrianMontaguSmartDCSIT) [reply tests/tel2puml/find_unique_graphs/test_json_data_converter.py]: Updated to string

**Review by FreddieMatherSmartDCSIT — APPROVED** [review]
Looks good to me

**Pull request merged**